pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -347,6 +350,23 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation) *gardencorev1
 			Fn:           flow.TaskFn(botanist.WaitUntilTunnelConnectionExists).SkipIf(o.Shoot.HibernationEnabled),
 			Dependencies: flow.NewTaskIDs(deployManagedResources, waitUntilNetworkIsReady, waitUntilWorkerReady),
 		})
+		_ = g.Add(flow.Task{
+			Name: "Migrating Kubernetes API server to ClusterIP service in the Seed cluster",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				return botanist.DeployKubeAPIService(ctx, sniPhase.Done())
+			}).
+				RetryUntilTimeout(defaultInterval, defaultTimeout).
+				SkipIf(o.Shoot.HibernationEnabled).
+				DoIf(sniPhase == component.PhaseEnabling || sniPhase == component.PhaseDisabling),
```

Comment:
If `sniPhase == component.PhaseDisabling` then we would want to have a load balancer service, right?

Reply:
The wording of the task was wrong. It should be 

```
Finishing Kubernetes API server service SNI transition
```


The idea is that kube-apiserver's service should be moved to the next phase.